### ISSUE TITLE
The request has been throttled on OneDrive

### ISSUE THREAD

**Issue by RyukSniper** [ncw/rclone#2406]
hi,
after I did the file upload, the folder did not respond like that then I restarted and now I receive this error.
I'm on a dedicated server with Ubuntu.
Failed to create file system for "Onedrive:": failed to get root: activityLimitReached: throttledRequest: The request has been throttled
how can i resolve?

**Comment by ncw** [comment]
You can experiment with `--tpslimit` to reduce the number of transactions per second.

You can also experiment with reducing `--checkers` and `--transfers` to say 1 and 2.

Here is microsoft's advice:

https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online

**Comment by ncw** [comment]
It might also be possible to improve things by decorating rclone's user agent as specified, so you could try using

    --user-agent "ISV|rclone|rclone/v1.42"

**Comment by RyukSniper** [comment]
ok thanks i try.

**Comment by zzzop** [comment]
Same problem on onedrive with `--checkers 1 --transfers 1 --tpslimit 1 --user-agent "ISV|rclone|rclone/v1.42"`

**Comment by diman82** [comment]
Same problem as @zzzop  - can't upload anything.

**Comment by RyukSniper** [comment]
I almost solved by uploading 1 file at a time with a time frame between a file and another of 1/2 hours

**Comment by ncw** [comment]
@Cnly you are probably right.  @olihey was working on Microsoft graph - not sure where he got to.

**Comment by Cnly** [comment]
Update: Without upgrading to Graph, I'm afraid that no new usable Client ID / Key could be created:

> Deprecation begins on January 10th, 2018. Newly registered apps will not be able to access the Office 365 discovery service. Existing apps can continue to use the service until November 1st, 2019. From November 1st, 2019 onward, the Office 365 discovery service will be fully decommissioned, and no apps will be able to use the service anymore.

(Source: https://developer.microsoft.com/en-us/office/blogs/outlook-rest-api-v1-0-office-365-discovery-and-live-connect-api-deprecation/)

**Comment by ncw** [comment]
@Cnly - I really appreciate your help on this - thanks :-) Looks like Graph is the way to go...

**Comment by RyukSniper** [comment]
how i sue Graph for rclone? or any update for integration in rclone?

**Comment by Cnly** [comment]
@RyukSniper You can probably find what you're looking for in this post https://forum.rclone.org/t/please-help-to-test-out-new-api-for-onedrive-less-throttling-better-sharepoint-support-etc/6570

**Comment by ncw** [comment]
Note that the graph support has been merged into [the latest beta](https://beta.rclone.org) now

**Comment by RyukSniper** [comment]
i download beta and try mount but i recive immediatly (error activityLimitReached: The request has been throttled), with new api is not resolve?

**Comment by Cnly** [comment]
@RyukSniper If you're using Office 365, there seems to be a major outage of the service now. I've got a lot of strange errors too. https://status.office.com/ Not sure if Personal version is affected.

Also, please note that config for OneDrive in v1.43 and beta are not compatible.

**Comment by RyukSniper** [comment]
@Cnly I use Office365 but I noticed that on OneDrive I enter it as a business, and from the web I can also have it as a sharepoint. so the errors are due to the version or the office problems?

**Comment by RyukSniper** [comment]
@Cnly ok thanks.

**Comment by RyukSniper** [comment]
@Cnly today on status.office.com no error, but rclone give me this error:
2018/09/05 12:28:10 Failed to query available drives: HTTP error 401 (401 Unauthorized) returned body: "{\r\n  \"error\": {\r\n    \"code\": \"InvalidAuthenticationToken\",\r\n    \"message\": \"Access token validation failure.\",\r\n    \"innerError\": {\r\n      \"request-id\": \"824f4779-7f98-4450-a933-bad302b12f16\",\r\n      \"date\": \"2018-09-05T12:28:10\"\r\n    }\r\n  }\r\n}"
i authorize in my pc and config  and paste token in a dedicated server, am I wrong? or is it because of latest version beta?

**Comment by Cnly** [comment]
@RyukSniper You mentioned that you authorized on your PC and pasted the token onto your server. What versions of rclone are you using for the two machines? Also, are you using custom Client ID / Key?

**Comment by RyukSniper** [comment]
@Cnly 
on server:rclone v1.43-039-gc6f1c3c7-beta
- os/arch: linux/amd64
- go version: go1.11
on pc:rclone v1.43
- os/arch: windows/amd64
- go version: go1.11
must also match the specific version?
and no i not using custom cliend id / key more than anything else I do not know what they are and then I leave white.
where can I get them if they are needed?

**Comment by Cnly** [comment]
@RyukSniper I think I saw your problem. rclone v1.43 has the legacy OneDrive API, and v1.43-xxx-**beta** are beta versions after the 1.43 release and they have the new API. You need to update rclone on your PC to a 1.43-beta version, maybe just the one on your server. Alternatively, you can get them here: https://beta.rclone.org/

**Comment by RyukSniper** [comment]
oh ok. now no error but on mount: 
rclone mount OneDrive: /mnt/OneDrive/ --vfs-cache-mode writes --allow-other -vv
2018/09/05 13:02:10 DEBUG : rclone: Version "v1.43-039-gc6f1c3c7-beta" starting with parameters ["rclone" "mount" "OneDrive:" "/mnt/OneDrive/" "--vfs-cache-mode" "writes" "--allow-other" "-vv"]
2018/09/05 13:02:10 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
2018/09/05 13:02:10 DEBUG : pacer: Rate limited, increasing sleep to 20ms
2018/09/05 13:02:10 DEBUG : pacer: low level retry 1/10 (error activityLimitReached: The request has been throttled)
2018/09/05 13:02:11 DEBUG : pacer: Rate limited, increasing sleep to 40ms
2018/09/05 13:02:11 DEBUG : pacer: low level retry 2/10 (error activityLimitReached: The request has been throttled)
2018/09/05 13:02:11 DEBUG : pacer: Rate limited, increasing sleep to 80ms
2018/09/05 13:02:11 DEBUG : pacer: low level retry 3/10 (error activityLimitReached: The request has been throttled)
2018/09/05 13:02:11 DEBUG : pacer: Rate limited, increasing sleep to 160ms
2018/09/05 13:02:11 DEBUG : pacer: low level retry 4/10 (error activityLimitReached: The request has been throttled)
2018/09/05 13:02:12 DEBUG : pacer: Reducing sleep to 120ms
2018/09/05 13:02:12 Fatal error: Can not open: /mnt/OneDrive/: open /mnt/OneDrive/: transport endpoint is not connected
how resolve?

**Comment by Cnly** [comment]
@RyukSniper That's because Microsoft is not generous on API usage. Please go through the section `Getting your own Client ID and Key` here: https://github.com/ncw/rclone/blob/master/docs/content/onedrive.md#getting-your-own-client-id-and-key

**Comment by Cnly** [comment]
@RyukSniper That's a little strange... I remember having seen this issue within only a short while after applying for my Client ID / Key. It's now been some time and does your problem continue to exist?

**Comment by RyukSniper** [comment]
yes. 
root@server:/mnt# rclone mount OneDrive: /mnt/OneDrive/ --vfs-cache-mode writes --allow-other -vv
2018/09/05 14:55:00 DEBUG : rclone: Version "v1.43-039-gc6f1c3c7-beta" starting with parameters ["rclone" "mount" "OneDrive:" "/mnt/OneDrive/" "--vfs-cache-mode" "writes" "--allow-other" "-vv"]
2018/09/05 14:55:00 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
2018/09/05 14:55:00 DEBUG : One drive root '': Token expired but no uploads in progress - doing nothing
2018/09/05 14:55:01 DEBUG : OneDrive: Saved new token in config file
2018/09/05 14:55:02 DEBUG : pacer: Rate limited, increasing sleep to 20ms
2018/09/05 14:55:02 DEBUG : pacer: low level retry 1/10 (error activityLimitReached: The request has been throttled)
2018/09/05 14:55:02 DEBUG : pacer: Rate limited, increasing sleep to 40ms
2018/09/05 14:55:02 DEBUG : pacer: low level retry 2/10 (error activityLimitReached: The request has been throttled)
2018/09/05 14:55:06 DEBUG : pacer: Reducing sleep to 30ms
2018/09/05 14:55:06 Fatal error: Can not open: /mnt/OneDrive/: open /mnt/OneDrive/: transport endpoint is not connected

from that fatal error I can not figure out whether it is due to my server or not.

**Comment by Cnly** [comment]
@RyukSniper Probably not with your server... But probably not rclone, either. I can't figure out the exact cause of your problem now. Are you the admin of your O365 and can you check if there are any settings potentially preventing this? Or maybe could you try with a OneDrive Personal backend?

**Comment by RyukSniper** [comment]
@Cnly i not secure admin my Office365, becouse this office365 for education. now i try with my personal OneDrive

**Comment by RyukSniper** [comment]
@Cnly with new graph api reduce throttling? becouse in a work day and hour day, my OneDrive only throttled :c

**Comment by RyukSniper** [comment]
Where can I contact to find some APIs?

**Comment by Cnly** [comment]
@RyukSniper I mean that you could contact your admin or Microsoft support and ask them why you're getting lots of throttling. Or maybe you can go to the [forum](https://forum.rclone.org/) and see if anybody else is having your problem - they may be able to help you. :)

**Comment by RyukSniper** [comment]
ok now resolve the api key, i re-create api, re-create on rlcone and now not throttling but for #2565 not resolve in console rlcone spam
2018/09/14 12:37:22 DEBUG : : Statfs:
2018/09/14 12:37:22 DEBUG : : >Statfs: stat={Blocks:1342177280 Bfree:1160372189 Bavail:1130009770 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
but resolve the view size. but why spam statfs?

**Comment by ncw** [comment]
@RyukSniper wrote
> @ncw but for #2565 not resolve in console rlcone spam
> 2018/09/14 12:37:22 DEBUG : : Statfs:
> 2018/09/14 12:37:22 DEBUG : : >Statfs: stat={Blocks:1342177280 Bfree:1160372189 Bavail:1130009770 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=
> but resolve the view size. but why spam statfs?

That will be your OS or some application which keeps doing stat() on the root of the file system.  I suggest you turn off the debug messages if you don't want to see those messages.

**Comment by Imparium** [comment]
Still getting this problem.

**Comment by Cnly** [comment]
I'm closing this because it contains outdated info for the legacy OneDrive API.

If you're still seeing similar problems, please ensure you are using rclone >= v1.44 (latest would be the best), and try getting your own client ID/key: https://github.com/ncw/rclone/blob/master/docs/content/onedrive.md#getting-your-own-client-id-and-key. Waiting for a while before retrying may also help.

**Comment by reinierpost** [comment]
I saw the same problem this morning with rclone 1.49.2 during an `rclone copy` command to my OneDrive Business drive on a 20 GB directory with over 140,000 files. I then upgraded to the current version (1.53.3) and restarted; thus far (an hour in) the problem hasn't reoccurred.

I'm here because I asked myself the exact same question: when throttling happens, how can I resolve it and successfully complete my action?

The above discussion makes it clear that we can twiddle with command line options and see if that helps, but in my use case, that is impractical: the command runs for hours, so it's not so easy to test various options, or to determine, when throttling happens, whether it could have been avoided in the first place. I need a reliable way to make sure all of those files are copied eventually and I need practical advice on how to achieve that.

Is specifying `--user-agent` still helpful or does rclone 1.53.3 do that for me now?

The man page mentions that getting and using an explicit Client ID may help, but this is not an option for us (my administrator has disabled it).
